src/types.jl

```diff
@@ -138,3 +138,16 @@ const ScalarType = Union{BitsType,Reference}
 
 # Define an H5O Object type
 const Object = Union{Group,Dataset,Datatype}
+
+idx_type(obj::File) =
+    get_context_property(:file_create).track_order ||
+    get_create_properties(obj).track_order ? 
+    API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
+idx_type(obj::Group) =
+    get_context_property(:group_create).track_order ||
+    get_create_properties(obj).track_order ? 
+    API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
+idx_type(obj) = API.H5_INDEX_NAME
+
+# TODO: implement alternative iteration order ?
+order(obj) = API.H5_ITER_INC
```

Comment:
I'd think a more descriptive name would be better. We might want to use `order` in the future for something else.

Reply:
Yes, left for another PR ;)